tools/win/generate_symbols.py

```diff
@@ -0,0 +1,289 @@
+"""Creates symsrv directory structure with executables and pdb files."""
+
+import argparse
+import asyncio
+import concurrent.futures
+import configparser
+import os
+import re
+import shutil
+import subprocess
+
+from datetime import datetime
+
+ROOT_DIR = os.path.join(os.path.dirname(__file__), *[os.pardir] * 3)
+
+
+async def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--installer-config',
+        required=True,
+        help='Installer config that contains files to process.',
+    )
+    parser.add_argument(
+        '--additional-files',
+        nargs='+',
+        help='Additional exe/dll files to process.',
+    )
+    parser.add_argument(
+        '--build-dir',
+        required=True,
+        help='Build directory.',
+    )
+    parser.add_argument(
+        '--toolchain-dir',
+        required=True,
+        help='Toolchain directory.',
+    )
+    parser.add_argument(
+        '--symbols-dir',
+        required=True,
+        help='Directory to output all files for symbol server.',
+    )
+    parser.add_argument(
+        '--pdb-only-symbols-dir',
+        help='Directory to output only pdb files.',
+    )
+    parser.add_argument(
+        '--run-source-index',
+        default=False,
+        action='store_true',
+        help="""
+          Run source_index.py from the Chromium toolset. This tool will take a
+          PDB on the command line, extract the source files that were used in
+          building the PDB, query the source server for which repository and
+          revision these files are at, and then finally write this information
+          back into the PDB in a format that the debugging tools understand.
+          This allows for automatic source debugging, as all of the information
+          is contained in the PDB, and the debugger can go out and fetch the
+          source files.
+        """,
+    )
+    parser.add_argument(
+        '--clear',
+        default=False,
+        action='store_true',
+        help='Clear the symbol directories before writing new symbols.',
+    )
+
+    args = parser.parse_intermixed_args()
+
+    if args.clear:
+        shutil.rmtree(args.symbols_dir, ignore_errors=True)
+        if args.pdb_only_symbols_dir:
+            shutil.rmtree(args.pdb_only_symbols_dir, ignore_errors=True)
+
+    tasks = [
+        run_with_semaphore(process_image(args, image_path))
+        for image_path in get_images_with_pdbs(args)
+    ]
+    for result in asyncio.as_completed(tasks):
+        print(await result)
+
+
+def get_images_with_pdbs(args):
+    images_with_pdbs = []
+
+    def add_if_has_pdb(file, should_exist=False):
+        file = os.path.join(args.build_dir, file)
+        if not os.path.exists(file):
+            if should_exist:
+                raise RuntimeError(f'{file} not found')
+            return
+        pdb_file = file + '.pdb'
+        if not os.path.exists(pdb_file):
+            return
```

Comment:
Is it a usual case? Or better to make it assert/RuntimeError ?

Reply:
It's an expected case. We process https://github.com/chromium/chromium/blob/116.0.5845.1/chrome/installer/mini_installer/chrome.release here, so we have `.pak`, `.json`, `.dat`, `.bin` files here.

Some `.dll`s may be prebuilt, so we can't assert `.pdb` existence either.

I'll add a sanity assert to make sure we have at least `chrome.dll.pdb`.